### PR TITLE
docs: add JSDocs for top level API objects

### PR DESCRIPTION
**Pull request by lance**
This commit adds JSDoc documentation to the `CloudEvent` and `HTTPReceiver`
objects exposed by the API when using the top level imports.

Ref: https://github.com/cloudevents/sdk-javascript/issues/130
Ref: https://github.com/cloudevents/sdk-javascript/issues/9

Creating as a draft PR because I'd still like to add docs generation for hosting on GitHub pages.

Signed-off-by: Lance Ball <lball@redhat.com>

### CODE REVIEW

**Comment by lance** [comment]
I didn't mark this as "Fixes" for #130 since I think we should discuss what the actual exposed API should be before we determine what constitutes acceptable documentation for end users. I think once a [version agnostic emitter](https://github.com/cloudevents/sdk-javascript/issues/124) is available, that should be exposed at the top level as well, and everything else should be removed from the public facing API. What else besides the `CloudEvent` and `HTTPEmitter` and an `HTTPReceiver` is needed for `CloudEvent` producers and consumers?

**Comment by lance** [comment]
I've added automatic doc generation. You can see how it will look on my fork here: http://lanceball.com/sdk-javascript/

File: lib/cloudevent.js [hunk 1]
```diff
@@ -1,24 +1,37 @@
...
+  /**
+   * Get the formatters available to this CloudEvent
+   * @returns {Object} a JSON formatter
+   */
   getFormats() {
     return { json: Formatter };
   }
 
+  /**
+   * Format the CloudEvent as JSON. Validates the event according
```
Comment (helio-frota) [inline lib/cloudevent.js:30]: Minor typo `Format` should be `Formats`.

File: lib/cloudevent.js [hunk 2]
```diff
@@ -30,81 +43,174 @@ class CloudEvent {
...
+   * @returns {*} any data set for this event
+   */
   getData() {
     return this.spec.getData();
   }
 
+  /**
+   * Adds an extension attribute to this CloudEvent
+   * @see https://github.com/cloudevents/spec/blob/master/spec.md#extension-context-attributes
+   * @param {*} key the name of the exteneion attribute
```
Comment (helio-frota) [inline lib/cloudevent.js:210]: Minor typo `exteneion`

**Comment by lance** [comment]
@helio-frota changes applied. Would you mind refreshing your review? Thanks.

File: docs/cloudevent.js.html [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+   * @returns {*} any data set for this event
+   */
+  getData() {
+    return this.spec.getData();
+  }
+
+  /**
+   * Adds an extension attribute to this CloudEvent
+   * @see https://github.com/cloudevents/spec/blob/master/spec.md#extension-context-attributes
+   * @param {*} key the name of the exteneion attribute
```
Comment (helio-frota) [inline docs/cloudevent.js.html:302]: @lance seems the typo was fixed in code but the generated docs needs to be updated again ?

Comment (helio-frota) [reply docs/cloudevent.js.html:302]: not a blocker since it will be updated along time anyway :+1:

Comment (lance) [reply docs/cloudevent.js.html:302]: Oh shoot - thanks. I'll update before landing this.

**Pull request merged**